File: dehaze-go/internal/model/sys_member_quota.go
package model

import "time"

type SysMemberQuota struct {
	ID            int64     `gorm:"primaryKey;autoIncrement;column:id" json:"id"`
	UserID        int64     `gorm:"column:user_id;type:bigint;not null;uniqueIndex:uk_user_month,priority:1;comment:用户ID" json:"userId"`
	QuotaMonth    int       `gorm:"column:quota_month;type:int;not null;uniqueIndex:uk_user_month,priority:2;index:idx_quota_month;comment:配额月份" json:"quotaMonth"`
	LevelCode     string    `gorm:"column:level_code;type:varchar(16);not null;comment:当月会员等级" json:"levelCode"`
	DehazeQuota   int       `gorm:"column:dehaze_quota;type:int;not null;default:0;comment:当月去雾配额" json:"dehazeQuota"`
	DehazeUsed    int       `gorm:"column:dehaze_used;type:int;not null;default:0;comment:当月已用去雾次数" json:"dehazeUsed"`
	EvaluateQuota int       `gorm:"column:evaluate_quota;type:int;not null;default:0;comment:当月评估配额" json:"evaluateQuota"`
	EvaluateUsed  int       `gorm:"column:evaluate_used;type:int;not null;default:0;comment:当月已用评估次数" json:"evaluateUsed"`
	ResetTime     time.Time `gorm:"column:reset_time;type:datetime;not null;comment:配额重置时间" json:"resetTime"`
	Deleted       int8      `gorm:"column:deleted;type:tinyint;not null;default:0;comment:逻辑删除标识" json:"-"`
	CreateTime    time.Time `gorm:"column:create_time;type:datetime;default:CURRENT_TIMESTAMP;comment:创建时间" json:"createTime"`
	UpdateTime    time.Time `gorm:"column:update_time;type:datetime;default:CURRENT_TIMESTAMP;comment:更新时间" json:"updateTime"`
	CreateBy      *int64    `gorm:"column:create_by;comment:创建人ID" json:"createBy"`
	UpdateBy      *int64    `gorm:"column:update_by;comment:修改人ID" json:"updateBy"`
}

func (SysMemberQuota) TableName() string {
	return "sys_member_quota"
}

File: dehaze-python/app/models/entity/sys_member_quota.py
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel, SoftDeleteMixin


class SysMemberQuota(BaseModel, SoftDeleteMixin):
    __tablename__ = "sys_member_quota"
    __table_args__ = {"comment": "会员月度配额历史表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment="用户ID")
    quota_month: Mapped[int] = mapped_column(Integer, nullable=False, comment="配额月份(yyyyMM)")
    level_code: Mapped[str] = mapped_column(String(16), nullable=False, comment="当月会员等级")
    dehaze_quota: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月去雾配额"
    )
    dehaze_used: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月已用去雾次数"
    )
    derain_quota: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月去雨配额"
    )
    derain_used: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月已用去雨次数"
    )
    desnow_quota: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月去雪配额"
    )
    desnow_used: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月已用去雪次数"
    )
    lowlight_quota: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月低光增强配额"
    )
    lowlight_used: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月已用低光增强次数"
    )
    super_resolution_quota: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月超分辨率配额"
    )
    super_resolution_used: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月已用超分辨率次数"
    )
    denoise_quota: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月去噪配额"
    )
    denoise_used: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月已用去噪次数"
    )
    inpaint_quota: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月图像修复配额"
    )
    inpaint_used: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月已用图像修复次数"
    )
    evaluate_quota: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月评估配额"
    )
    evaluate_used: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="当月已用评估次数"
    )
    reset_time: Mapped[datetime] = mapped_column(DateTime, nullable=False, comment="配额重置时间")
